src/parser.rs

```diff
@@ -816,6 +824,18 @@ impl<'tokens, 'src> Parser<'tokens, 'src> {
 
     Ok(Shell { arguments, command })
   }
+
+  /// Parse a recipe attribute
+  fn parse_attribute(&mut self) -> CompileResult<'src, Attribute> {
+    let token = self.expect(Identifier)?;
+    let lexeme = token.lexeme();
+    if let Some(attribute) = Attribute::from_lexeme(lexeme) {
+      self.expect(BracketR)?;
+      Ok(attribute)
+    } else {
+      Err(token.error(CompileErrorKind::UnknownAttribute { attribute: lexeme }))
+    }
+  }
```

Comment:
This is not quite right, since you also need the token to produce the error message, but something like this:

```suggestion
  /// Parse a recipe attribute name
  fn parse_attribute_name(&mut self) -> CompileResult<'src, Attribute> {
    let name = self.parse_name()?;
    Attribute::from_lexeme(name).ok_or_else(|| token.error(CompileErrorKind::UnknownAttribute { attribute: lexeme }))
    }
```